deploy/cdk/lib/maintain-metadata/maintain-metadata-stack.ts

```diff
@@ -1124,6 +1124,93 @@ def _delete_expired_api_keys(graphql_api_id: str):
+            #set($portfolioCollectionId = $util.defaultIfNullOrBlank($portfolioCollectionId, ""))
+            #set($portfolioCollectionId = $util.str.toUpper($portfolioCollectionId))
+            #set($portfolioCollectionId = $util.str.toReplace($portfolioCollectionId, " ", ""))
+
+            #set($privacy = $util.defaultIfNullOrBlank($ctx.args.privacy, "private"))
+            #if( $util.isBoolean($ctx.args.featuredCollection) )
+              #set($featuredCollection = $ctx.args.featuredCollection)
+            #else
+              #set($featuredCollection = false)
+            #end
+            #if( $util.isBoolean($ctx.args.highlightedCollection) )
+              #set($highlightedCollection = $ctx.args.highlightedCollection)
+            #else
+              #set($highlightedCollection = false)
+            #end
+
+            #if( $privacy != "public")
+              #set($featuredCollection = false)
+              #set($highlightedCollection = false)
+            #end
+
+            #set($layout = $util.defaultIfNullOrBlank($ctx.args.layout, "default"))
+
+            #set($title = $util.defaultIfNullOrBlank($ctx.args.title, $null))
+
+            #set($pk = "PORTFOLIO")
+            #set($sk = $util.str.toUpper("USER#$portfolioUserId#$portfolioCollectionId"))
+
+            #set( $expValues = {})
+            $!{expValues.put(":portfolioCollectionId", $util.dynamodb.toDynamoDB($portfolioCollectionId))}
+            $!{expValues.put(":portfolioUserId", $util.dynamodb.toDynamoDB($portfolioUserId))}
+            $!{expValues.put(":rowType", $util.dynamodb.toDynamoDB("PortfolioCollection"))}
+            $!{expValues.put(":dateAddedToDynamo", $util.dynamodb.toDynamoDB($util.time.nowISO8601()))}
+            $!{expValues.put(":dateModifiedInDynamo", $util.dynamodb.toDynamoDB($util.time.nowISO8601()))}
+            $!{expValues.put(":featuredCollection", $util.dynamodb.toDynamoDB($featuredCollection))}
+            $!{expValues.put(":highlightedCollection", $util.dynamodb.toDynamoDB($highlightedCollection))}
+            $!{expValues.put(":privacy", $util.dynamodb.toDynamoDB($privacy))}
+            $!{expValues.put(":GSI1PK", $util.dynamodb.toDynamoDB("PORTFOLIOCOLLECTION"))}
+            $!{expValues.put(":GSI1SK", $util.dynamodb.toDynamoDB("PORTFOLIOCOLLECTION#$portfolioCollectionId"))}
+
+            #if( $privacy == "private" )
+              #set($GSI2PK = "")
+              #set($GSI2SK = "")
+            #else
+              #set($GSI2PK = "PORTFOLIOCOLLECTION")
+              #set($GSI2SK = $util.str.toUpper("$privacy#$portfolioCollectionId"))
+
+              $!{expValues.put(":GSI2PK", $util.dynamodb.toDynamoDB($GSI2PK))}
+              $!{expValues.put(":GSI2SK", $util.dynamodb.toDynamoDB($GSI2SK))}
+            #end
+
+            {
+              "version": "2017-02-28",
+              "operation": "UpdateItem",
```

Comment:
You may want to double-check that other existing fields not specified in the update statement are retained.  I seem to recall that if something isn't specified, it will get nulled out.  You might want to verify.

Reply:
Yes, they are retained.